templates/todo/projects/java-mongo-aca/.repo/bicep/infra/main.bicep

```diff
@@ -56,147 +94,346 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+        roleDefinitionIdOrName: acrPullRole
+      }
+    ]
+  }
+}
+
+//Container apps environment
+module containerAppsEnvironment 'br/public:avm/res/app/managed-environment:0.4.5' = {
+  name: 'container-apps-environment'
+  scope: rg
+  params: {
+    logAnalyticsWorkspaceResourceId: logAnalytics.outputs.resourceId
+    name: !empty(containerAppsEnvironmentName) ? containerAppsEnvironmentName : '${abbrs.appManagedEnvironments}${resourceToken}'
+    location: location
+    zoneRedundant: false
+    tags: tags
+  }
+}
+
+//the managed identity for web frontend
+module webIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.2.1' = {
+  name: 'webidentity'
+  scope: rg
+  params: {
+    name: '${abbrs.managedIdentityUserAssignedIdentities}web-${resourceToken}'
+    location: location
   }
 }
 
 // Web frontend
-module web '../../../../../common/infra/bicep/app/web-container-app.bicep' = {
+module web 'br/public:avm/res/app/container-app:0.2.0' = {
   name: 'web'
   scope: rg
   params: {
     name: !empty(webContainerAppName) ? webContainerAppName : '${abbrs.appContainerApps}web-${resourceToken}'
+    containers: [
+      {
+        image: 'mcr.microsoft.com/azuredocs/containerapps-helloworld:latest'
+        name: 'simple-hello-world-container'
+        resources: {
+          cpu: json('0.5')
+          memory: '1.0Gi'
+        }
+      }
+    ]
+    managedIdentities:{
+      systemAssigned: false
+      userAssignedResourceIds: [webIdentity.outputs.resourceId]
+    }
+    registries:[
+      {
+        server: '${containerRegistry.outputs.name}.${containerRegistryHostSuffix}'
+        identity: webIdentity.outputs.resourceId
+      }
+    ]
+    dapr: {
+        enabled: true
+        appId: 'main'
+        appProtocol: 'http'
+        appPort: 80
+    }
+    environmentId: containerAppsEnvironment.outputs.resourceId
+    location: location
+    tags: union(tags, { 'azd-service-name': 'web' })
+  }
+}
+
+//the managed identity for api backend
+module apiIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.2.1' = {
+  name: 'apiidentity'
+  scope: rg
+  params: {
+    name: '${abbrs.managedIdentityUserAssignedIdentities}api-${resourceToken}'
     location: location
-    tags: tags
-    identityName: '${abbrs.managedIdentityUserAssignedIdentities}web-${resourceToken}'
-    containerAppsEnvironmentName: containerApps.outputs.environmentName
-    containerRegistryName: containerApps.outputs.registryName
-    containerRegistryHostSuffix: containerRegistryHostSuffix
-    exists: webAppExists
   }
 }
 
 // Api backend
-module api '../../../../../common/infra/bicep/app/api-container-app.bicep' = {
+module api 'br/public:avm/res/app/container-app:0.2.0' = {
   name: 'api'
   scope: rg
   params: {
     name: !empty(apiContainerAppName) ? apiContainerAppName : '${abbrs.appContainerApps}api-${resourceToken}'
+    containers: [
+      {
+        image: 'mcr.microsoft.com/azuredocs/containerapps-helloworld:latest'
```

Comment:
if you are hard coding the helloworld image here then if a user does an `azd provision` after an `azd deploy` it will reset the container app back to helloworld.

Our upsert containerapp module handled this inconsistency.